### ISSUE TITLE
Estimators should comply with scikit-learn standards

### ISSUE THREAD

**Issue by popescu-v** [KhiopsML/khiops-python#87]
According to the developer guide of [Scikit-Learn][1] estimator classes should pass the `check_estimator` function. 

Some important things to change:
- Type and value checks should be done only when calling `fit` (Done already)
- Implement sklearn "Attributes" (data available post `fit`)
 - Note that they are `@property`
 - Examples: `n_classes_`, `n_features_in_`
 - Our specific data should be available in this form as well: `report_`, `khiops_model_`.

### Implementation Ideas
- Make a test that checks every estimator with `check_estimator` function of sklearn
 - Not sure it will work because our `X`'s are non-standard. Maybe test only with monotable?
- Debug until it passes

[1]: https://scikit-learn.org/stable/developers/develop.html#rolling-your-own-estimator

**Comment by popescu-v** [comment]
Part of this has been resolved when solving #156 because now all estimators return `numpy.ndarray`

**Comment by lucaurelien** [comment]
I've encountered an issue while integrating Khiops-Python into a Sklearn pipeline using _cross_val_score_. The method works fine for most metrics but fails when the metric is set to "roc_auc", returning NaN values.

To reproduce it, here is the code I've used:
`from sklearn.model_selection import cross_val_score
scores = cross_val_score(pkc, X, y, cv=5, scoring='roc_auc') 
print("cross-val scores:", scores)`

it returns **[nan nan nan nan nan]**

For instance, it works using 'accuracy':
`from sklearn.model_selection import cross_val_score
scores = cross_val_score(pkc, X, y, cv=5, scoring='accuracy') 
print("cross-val scores:", scores)`
Output: [0.86948511 0.87071348 0.87479525 0.86936937 0.87162162]

This issue affects benchmarking in PyCaret, as the resulting table contains 0.0 score for AUC.
